File: SuncoLab.Frontend/src/app/models/blog.ts
import { Image } from "./image";

export class Blog {
    id: string = '';
    name: string = '';
    description?: string;
    body: string = '';
    show: boolean = true;
    coverImage?: Image;

    constructor (id: string, name: string, html: string, show: boolean, description: string|undefined, coverImage: Image) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.body = html;
        this.show = show;
        this.coverImage = coverImage;
    }
}

File: SuncoLab.Model/Database/Blog.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace SuncoLab.Model
{
    public class Blog : BaseEntity
    {
        [Column(TypeName = "text")]
        public string Body { get; set; }
        public string Name { get; set; }
        public string? Description { get; set; }
        public bool Show { get; set; } = true;

        [ForeignKey("CoverImage")]
        public Guid? CoverImageId { get; set; }
        public Image CoverImage { get; set; }
    }
}
